accounting/src/main/java/io/skai/accounting/repository/impl/ModelRepositoryImpl.java

```diff
@@ -33,11 +38,34 @@ public List<Model> findAll(Long brandId) {
                 .fetchInto(Model.class);
     }
 
-    private ModelRecord createNewModel(Long brandId, String name){
+    @Override
+    public ModelInfoDto findModelInfo(Long id) {
+        return dslContext.select()
+                .from(MODEL)
+                .innerJoin(BRAND).on(MODEL.BRAND_ID.eq(BRAND.ID))
+                .where(MODEL.ID.eq(id))
+                .fetch(this::recordToImmutablePair)
+                .stream()
+                .map(this::mapToModelInfo)
+                .findFirst()
+                .orElseThrow();
+    }
+
+    private ModelRecord createNewModel(Long brandId, String name) {
         return dslContext.insertInto(MODEL)
                 .set(MODEL.BRAND_ID, brandId)
                 .set(MODEL.NAME, name)
                 .returning()
                 .fetchOne();
     }
+
+    private ModelInfoDto mapToModelInfo(ImmutablePair<Brand, Model> pair) {
```

Comment:
naming in such way is not appropriate because you do not map (mutate) you create, in such cases toModelInfo is a better and more common name